utils/duration_extractor.py

```diff
@@ -0,0 +1,130 @@
+from typing import Tuple
+
+import numpy as np
+import torch
+from scipy.sparse import coo_matrix, csr_matrix
+from scipy.sparse.csgraph import dijkstra
+
+from utils.text.symbols import silent_phonemes_indices
+
+
+def to_node_index(i: int, j: int, cols: int) -> int:
+    return cols * i + j
+
+
+def from_node_index(node_index: int, cols: int) -> Tuple[int, int]:
+    return node_index // cols, node_index % cols
+
+
+def to_adj_matrix(mat: np.array) -> csr_matrix:
```

Comment:
Do you think could be a good idea to move them as a class private methods? Or they are planned to be used from outside of the class as well?